src/loaddb/load_server_loader.cpp

```diff
@@ -675,6 +675,24 @@ namespace cubload
 	return;
       }
 
+    if (m_scancache.m_index_stats != NULL)
+      {
+	for (const auto &it : m_scancache.m_index_stats->get_map ())
+	  {
+	    if (!it.second.is_unique ())
```

Comment:
Probably a safe-guard `assert (it.second.is_unique ());` is enough.

Reply:
I think that we actually need to throw an error here, unique constraint violation, and abort the load as well.